pina/model/layers/rbf_layer.py

```diff
@@ -0,0 +1,323 @@
+    return -torch.sqrt(r**2 + 1)
+
+def inverse_multiquadric(r):
+    return 1/torch.sqrt(r**2 + 1)
+
+def inverse_quadratic(r):
+    return 1/(r**2 + 1)
+
+def gaussian(r):
+    return torch.exp(-r**2)
+
+RADIAL_FUNCS = {
+   "linear": linear,
+   "thin_plate_spline": thin_plate_spline,
+   "cubic": cubic,
+   "quintic": quintic,
+   "multiquadric": multiquadric,
+   "inverse_multiquadric": inverse_multiquadric,
+   "inverse_quadratic": inverse_quadratic,
+   "gaussian": gaussian
+   }
+
+SCALE_INVARIANT = {"linear", "thin_plate_spline", "cubic", "quintic"}
+
+MIN_DEGREE = {
+    "multiquadric": 0,
+    "linear": 0,
+    "thin_plate_spline": 1,
+    "cubic": 1,
+    "quintic": 2
+    }
+
+
+class RBFLayer(torch.nn.Module):
+    """
+    Radial Basis Function (RBF) interpolation layer.
+
+    Parameters
+    ----------
+    neighbors : int or None
+        Number of neighbors to use in the interpolation. If None, all points
+        are used.
+    smoothing : float or tensor
+        Smoothing parameter for the RBF interpolation. If a scalar, the same
+        smoothing is used for all points. If a tensor, it must have the same
+        length as the first dimension of `y`.
+    kernel : str
+        Radial basis function to use. Must be one of "linear", "thin_plate_spline",
+        "cubic", "quintic", "multiquadric", "inverse_multiquadric", "inverse_quadratic",
+        or "gaussian".
+    epsilon : float
+        Scaling parameter for the kernel. Must be specified for some kernels.
+    degree : int or None
+        Degree of the polynomial to include in the interpolation. If None, the
+        degree is chosen automatically.
+    device : str
+        Device on which to perform calculations. Default is "cpu".
+
+    Attributes
+    ----------
+    y : (n, d) tensor
+        Data points.
+    d : (n, m) tensor
+        Data values.
+    smoothing : (n,) tensor
+        Smoothing parameter for each data point.
+    kernel : str
+        Radial basis function to use.
+    epsilon : float
+        Scaling parameter for the kernel.
+    degree : int
+        Degree of the polynomial to include in the interpolation.
+    device : str
+        Device on which to perform calculations.
+    powers : (r, d) tensor
+        Powers for each monomial in the polynomial.
+    _shift : (d,) tensor
+        Shift for the data.
+    _scale : (d,) tensor
+        Scale for the data.
+    _coeffs : (n + r, m) tensor
+        Coefficients for the RBF interpolation.
+
+    """
+    def __init__(
+        self,
+        neighbors=None,
+        smoothing=0.0,
+        kernel="thin_plate_spline",
+        epsilon=None,
+        degree=None,
+        device="cpu",
+    ):
+        super().__init__()
+
+        self.device = device
+        self.smoothing = smoothing
+        self.neighbors = neighbors
+        self.kernel = kernel
+        self.epsilon = epsilon
+        self.degree = degree
+
+    def _check_data(self, y, d):
+        if y.ndim != 2:
```

Comment:
Can you comment on this: what about if we batch and we have an input `[B, Npoints, d]`?

Reply:
The input data to be fitted by the RBF should have dimension `(npoints, ndim)` and `(npoints, ...)` (as in the original class `scipy.interpolate.RBFInterpolator`)